src/os_db.ml

```diff
@@ -158,294 +165,330 @@ let preregister_table =
 
 (*****************************************************************************)
 
+module Utils = struct
+
+  let ( => ) v (f, g) = match v with
+    | Some v -> f v
+    | None -> g v
+
+  let as_sql_string v = <:value< $string:v$>>
+
+  let run_query q = full_transaction_block (fun dbh ->
+    Lwt_Query.query dbh q)
+  
+  let run_view q = full_transaction_block (fun dbh ->
+    Lwt_Query.view dbh q)
+
+  let run_view_opt q = full_transaction_block (fun dbh ->
+    Lwt_Query.view_opt dbh q)
+
+  let one f ~success ~fail q =
+    f q >>= function
+    | r::_ -> success r
+    | _ -> fail
+
+  let all f ~success ~fail q =
+    f q >>= function
+    | [] -> fail
+    | r -> success r
```

Comment:
To be honest, I do not like `one` and `all`. We don't save much code (given that we have to lambda-abstract the bodies of `success`, `fail`, and `f q`) and it is not immediately visible what success and fail mean (it is more like found and not-found).